### ISSUE TITLE
linter: stylish formatter reports wrong source code position

### ISSUE THREAD

**Issue by hi-ogawa** [oxc-project/oxc#9098]
### What version of Oxlint are you using?

0.15.10

### What command did you run?

pnpm dlx oxlint -D no-debugger --format stylish repro.js

### What does your `.oxlintrc.json` config file look like?

Probably none. I run the above command from oxc repo root.

### What happened?

The issue was mentioned in the comment https://github.com/oxc-project/oxc/issues/8422#issuecomment-2605064555, but this didn't seem to get addressed yet.

The code of `repro.js` is:

```js
debugger
```

The command output is a following:

```sh
# NOT OK
$ pnpm dlx oxlint -D no-debugger --format stylish repro.js

/home/hiroshi/code/others/oxc/repro.js
  0:8  error  `debugger` statement is not allowed  eslint(no-debugger)


# OK
$ pnpm dlx oxlint -D no-debugger --format unix repro.js
repro.js:1:1: `debugger` statement is not allowed [Error]


# OK
$ pnpm dlx oxlint -D no-debugger repro.js

  × eslint(no-debugger): `debugger` statement is not allowed
   ╭─[repro.js:1:1]
 1 │ debugger
...


# OK
$ pnpm dlx eslint  --no-config-lookup --rule 'no-debugger: error' --format stylish repro.js

/home/hiroshi/code/others/oxc/repro.js
  1:1  error  Unexpected 'debugger' statement  no-debugger
```
